gnss_lib_py/parsers/precise_ephemerides.py

```diff
@@ -0,0 +1,290 @@
+"""Functions to process precise ephemerides .sp3 and .clk files.
+
+"""
+
+__authors__ = "Sriramya Bhamidipati"
+__date__ = "09 June 2022"
+
+import os
+import warnings
+
+from datetime import datetime, timezone
+import numpy as np
+
+from gnss_lib_py.utils.time_conversions import datetime_to_gps_millis
+
+# Define the number of sats to create arrays for
+NUMSATS = {'gps': (32, 'G'),
+           'galileo': (36, 'E'),
+           'beidou': (46, 'C'),
+           'glonass': (24, 'R'),
+           'qzss': (3, 'J')}
+
+class Sp3:
+    """Class handling satellite position data from precise ephemerides
+
+    """
+    def __init__(self):
+        self.const = None
+        self.xpos = []
+        self.ypos = []
+        self.zpos = []
+        self.tym = []
+        self.utc_time = []
+
+    def __eq__(self, other):
+        """Checks if two Sp3() classes are equal to each other
+
+        Parameters
+        ----------
+        other : gnss_lib_py.parsers.precise_ephemerides.Sp3
+            Sp3 object that stores .sp3 parsed information
+
+        Returns
+        ----------
+        bool_check : bool
+            Flag (True/False) that indicates if Sp3 classes are equal
+        """
+        bool_check = (self.const == other.const) & \
+                     (self.xpos == other.xpos) & \
+                     (self.ypos == other.ypos) & \
+                     (self.zpos == other.zpos) & \
+                     (self.tym == other.tym) & \
+                     (self.utc_time == other.utc_time)
+
+        return bool_check
+
+def parse_sp3(input_path, constellation = 'gps'):
+    """sp3 specific loading and preprocessing for any GNSS constellation
+
+    Parameters
+    ----------
+    input_path : string
+        Path to sp3 file
+    constellation : string
+        Key from among {gps, galileo, glonass, beidou, qzss, etc} that
+        specifies which GNSS constellation to be parsed from .sp3 file
+        (the default is 'gps')
+
+    Returns
+    -------
+    sp3data : np.ndarray
+        Array of gnss_lib_py.parsers.precise_ephemerides.Sp3 with len = NUMSATS,
```

Comment:
List not array

Reply:
done, cross-checked test_precise_ephemerides as well. fix all such instances.